worldengine/common.py

```diff
@@ -120,3 +120,33 @@ def array_to_matrix(array, width, height):
         for x in range(width):
             matrix[y].append(array[y * width + x])
     return matrix
+
+def _equal(a, b):
+    #recursion on subclasses of types: tuple, list, dict
+    #specifically checks             : float, ndarray
```

Comment:
On second thought the removal would leave the world-class without a functioning equality-operator. At the same time it is not only usable for that class... I wouldn't feel good about moving this to the tests, I prefer it here. (I found quite a lot of functions spread over various files that are never called or even referenced... maybe it is time for a bit of cleanup anyway.)